### PR TITLE
Fixes #165 Hiding sidebar when no entity is selected

### LINKED ISSUE #165: Consider hiding "Attributes" sidebar when no scene-graph entity is selected
1. Load a scene with the editor mode.
2. Notice an empty "Attributes" sidebar appears:

image

### PR DESCRIPTION
**Pull request by fernandojsg**
hidesidebar

### CODE REVIEW

File: src/components/components/Sidebar.js [hunk 1]
```diff
@@ -31,24 +31,24 @@ export default class Sidebar extends React.Component {
...
+            {entityButtons}
+          </div>
+          <ComponentsContainer entity={this.props.entity}/>
         </div>
-        <ComponentsContainer entity={this.props.entity}/>
-      </div>
-    );
+      );
+    } else {
+      return <div></div>
```
Comment (ngokevin) [inline src/components/components/Sidebar.js]: Could do `<div/>`

**Comment by ngokevin** [comment]
There's no way to hide the sidebar after initial entity select (e.g., clicking on empty space of scene graph, clicking on scene, close panel option).

**Comment by fernandojsg** [comment]
I'm fixing these issues, but one is already on the way (selecting a-scene) on this PR: https://github.com/aframevr/aframe-inspector/pull/230

**Comment by fernandojsg** [comment]
@ngokevin I've made some changes that should works on all the cases except while clicking on a-scene, could you give it a try?

**Comment by fernandojsg** [comment]
Updated after merging a dependent PR (https://github.com/aframevr/aframe-inspector/pull/230)

**Comment by fernandojsg** [comment]
@ngokevin could you take a look at this one?

File: src/lib/inspector.js [hunk 2]
```diff
@@ -179,15 +179,17 @@ Inspector.prototype = {
...
-        if (mutation.type === 'childList') {
-          Array.prototype.slice.call(mutation.removedNodes).forEach(function (removedNode) {
-            if (self.selectedEntity === removedNode) {
-              self.selectEntity(null);
-            }
-          });
-        }
-      });
+      if (mutations) {
+        mutations.forEach((mutation) => {
```
Comment (ngokevin) [inline src/lib/inspector.js]: Don't need parens if single-arg

File: src/lib/inspector.js [hunk 3]
```diff
@@ -179,15 +179,17 @@ Inspector.prototype = {
...
+          if (mutation.type === 'childList') {
+            Array.prototype.slice.call(mutation.removedNodes).forEach(function (removedNode) {
```
Comment (ngokevin) [inline src/lib/inspector.js]: Don't need `self` if you use arrow binding syntax

File: src/lib/inspector.js [hunk 4]
```diff
@@ -179,15 +179,17 @@ Inspector.prototype = {
...
-      mutations.forEach((mutation) => {
-        if (mutation.type === 'childList') {
-          Array.prototype.slice.call(mutation.removedNodes).forEach(function (removedNode) {
-            if (self.selectedEntity === removedNode) {
-              self.selectEntity(null);
-            }
-          });
-        }
-      });
+      if (mutations) {
```
Comment (ngokevin) [inline src/lib/inspector.js]: If you return early, you could reduce the indent of the block below

`if (!mutations) { return; }`

`if (mutation.type !=='childList') { return; }`

**Comment by ngokevin** [comment]
r+wc
- Later would be nice to add a transition to the panel hide.
- I assume clicking on scene graph empty space to hide panel is fixed too.

Just to note, here's how MagicaVoxel does it. Don't have to do it this way, I think this pattern only works better if there are multiple panels:

magica

File: src/lib/inspector.js (hunk 3 above)
Comment (fernandojsg) [inline src/lib/inspector.js]: Yep, missed that, thanks!

File: src/lib/inspector.js (hunk 2 above)
Comment (fernandojsg) [inline src/lib/inspector.js]: Yep, fixed!

File: src/lib/inspector.js (hunk 4 above)
Comment (fernandojsg) [inline src/lib/inspector.js]: Ok, changed!

**Comment by fernandojsg** [comment]
Ok ,thank you. I'll file issues with the proposed changes

**Pull request merged**